classes/03class/aws/README.md

```diff
@@ -0,0 +1,184 @@
+### Application Load Balancer (ALB)
+
+The service can be found at Services > Compute > EC2 > Load Balancing > Load Balancers > Create Load Balancer > ALB
+
+An ALB can serve as a single point of contact for clients and distribute traffic across multiple targets, such as EC2 Instances, in multiple Availability Zones. When creating an ALB you will have to chose from 2 different types and in which Availability Zones your LB will be available:
+
+- internet-facing: routes requests from public internet to targets
+- internal: routes requests only from clients to targets using private IP addresses
+
+Basically an ALB has two main components: listeners and target groups.
+
+```Load balancers are included on the AWS free-tier. However, please remember to delete any unused resources to avoid any costs```
+
+#### listeners
+
+A listener tries to match requests from clients based on, among many types, protocol/port and path-based routing. For example a request on port `443` can redirect the traffic to an internal server running on port `80` or a request to `/admin` sends the traffic to a specifc EC2 Instance.
+
+Each listener rule consists in a priority of actions and one or more conditions. When the conditions for a rule are met, the traffic is then forwarded to a target group.
+
+#### target group
+
+Target groups send requests to one EC2, an ASG and others. Healthchecks can be used to check whether the service is up and running and can start receiving traffic from the load balancer.
+
+Those healthchecks can also be linked to an ASG to send termination requests if the instance is not working correctly.
+
+![alb](/classes/03class/aws/assets/alb.png)
+
+#### AWS Certificate Manager
+
+If your service is being exposed through an ALB on port 443, a SSL certificate will need to be attached to the listener, otherwise when your clients try to access your webserver, their browser will complain about the connection not being safe.
+
+AWS offers a service for generating SSL certificates for free if you are using with their managed services like a load balancer.
+
+Read more about AWS ACM and SSL certificates [here](https://aws.amazon.com/certificate-manager/getting-started/?nc=sn&loc=7)
+
+## Route53
+
+R53 is the DNS managed service offered by AWS. To consume this service you first will need to create either private or public hosted zones.
+
+- public hosted zone: can be public accessible. You will need to own a domain and point the `ns` records to the zone created to be managed by AWS.
+- private hosted zone: will only be resolved by local workloads running on your VPC.
+
+Unfortunately this is not included on the AWS free-tier. However the cost is only $1AUD/month per hosted zone.
+
+If you wish to try a public hosted zone, you can generate a free domain at `http://www.dot.tk/en/index.html?lang=en`, create a public hosted zone with the same name as your domain and follow the directions to create the `ns` records and managed your new domain through Route53.
+
```

Comment:
Include different types of targets R53 can handle, including ALB alias and some basic routing options.

Reply:
I reckon you should also should mention record types or have a link to an article explaining the most common ones(at least A and CNAME). Even having the link in the appendix, it's good to mention it on this section and reference the link below